From 6d85832a7913888b3538b742c3807a57c24085d6 Mon Sep 17 00:00:00 2001
From: Jon Lin <jon.lin@rock-chips.com>
Date: Wed, 23 Dec 2020 14:27:07 +0800
Subject: [PATCH] drivers: rkflash: Remove RK vendor support for spinor mtd
 case

1.Change to use RK MTD vendor operation
2.RK MTD vendor is incompatible with RK vendor

Change-Id: I7c233b0b0a98c5e93d0722956809a9d6c01663a1
Signed-off-by: Jon Lin <jon.lin@rock-chips.com>
---
 drivers/rkflash/rkflash_blk.c | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/drivers/rkflash/rkflash_blk.c b/drivers/rkflash/rkflash_blk.c
index b085501b9e07..6030ac905301 100644
--- a/drivers/rkflash/rkflash_blk.c
+++ b/drivers/rkflash/rkflash_blk.c
@@ -624,7 +624,7 @@ static void rkflash_blk_unregister(struct flash_blk_ops *blk_ops)
 	unregister_blkdev(blk_ops->major, blk_ops->name);
 }
 
-static int rkflash_dev_vendor_read(u32 sec, u32 n_sec, void *p_data)
+static int __maybe_unused rkflash_dev_vendor_read(u32 sec, u32 n_sec, void *p_data)
 {
 	int ret;
 
@@ -639,7 +639,7 @@ static int rkflash_dev_vendor_read(u32 sec, u32 n_sec, void *p_data)
 	return ret;
 }
 
-static int rkflash_dev_vendor_write(u32 sec, u32 n_sec, void *p_data)
+static int __maybe_unused rkflash_dev_vendor_write(u32 sec, u32 n_sec, void *p_data)
 {
 	int ret;
 
@@ -682,8 +682,12 @@ int rkflash_dev_init(void __iomem *reg_addr,
 	/* vendor part */
 	switch (type) {
 	case FLASH_TYPE_SFC_NOR:
+#if IS_ENABLED(CONFIG_RK_SFC_NOR_MTD) && IS_ENABLED(CONFIG_ROCKCHIP_MTD_VENDOR_STORAGE)
+		break;
+#else
 		flash_vendor_dev_ops_register(rkflash_dev_vendor_read,
 					      rkflash_dev_vendor_write);
+#endif
 		break;
 	case FLASH_TYPE_SFC_NAND:
 #ifdef CONFIG_RK_SFC_NAND_MTD
-- 
2.35.3

